### DISCUSSION TITLE
Add element tools to UML

### DISCUSSION THREAD

**Discussion by aylafiryal** [clientIO/joint#2216] [Q&A] [answered]
### Introduction

I'm want to add a remove button to my class diagram through this code:

`var classes = {mammal : new uml.Interface(), person: new uml.Abstract()} //list of the classes`

`var removeButton = new joint.elementTools.Remove();`
`var toolsView = new joint.dia.ToolsView({`
`    tools: [`
`        removeButton`
`    ]`
`});`

`Object.values(classes).forEach(function(obj) {`
`        var elementView = obj.findView(paper);`
`        elementView.addTools(toolsView);`
`        paper.on('element:mouseenter', function(elementView) { // In the video I provide, I haven't implemented this code yet, but this code runs well`
`                elementView.showTools();`
`        });`
`        paper.on('element:mouseleave', function(elementView) {`
`                elementView.hideTools();`
`        });`
`});`

But apparently, the tools disappear if a new class is created (meaning that the tools only show right after a class is created and if I create another one next, the tools are gone from the last class). I think it's because I use forEach loop. But based on the demo, I have to use an object like `classes` so in order to add tools, I need to loop each of the values. 

How can I make the tools always appear and don't disappear in old classes when new classes are created?

https://github.com/clientIO/joint/assets/83856972/4e3db54a-75b6-4832-a7c5-f8e5a05af3a7

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
It can be done simply as this:

```js
const removeButton = new joint.elementTools.Remove();
const toolsView = new joint.dia.ToolsView({
    tools: [
        removeButton
    ]
});

paper.on('element:mouseenter', function(elementView) {
    elementView.addTools(toolsView);
});
paper.on('element:mouseleave', function(elementView) {
    elementView.removeTools();
});
```

**Reply by aylafiryal** [reply]
Thank you for the answer, this works perfectly! Really appreciate it. However, I noticed that my second question wasn't addressed. I'm still curious about [my other question](https://github.com/clientIO/joint/discussions/2212). Would you mind taking a moment to provide an answer to that as well?
